### PR TITLE
Application metrics and performance tests for Flow operations

### PR DESCRIPTION
**Pull request by sergii-iakovenko**
The changes:
- The implementation of application metrics stored in OpenTSDB.
- Introduce metrics for Flow operations.
- Performance tests for basic Flow operations.

Related to #3605

### CODE REVIEW

File: src-java/testing/performance-tests/src/test/groovy/org/openkilda/performancetests/spec/benchmark/FlowDumpSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            flowHelperV2.addFlow(flow)
+            flows << flow
+        }
+
+        and: "Flows are created"
+        assert flows.size() == preset.flowCount
+
+        then: "Dump flows"
+        (1..preset.dumpAttempts).each {
+            northboundV2.getAllFlows().size() == preset.flowCount
```
Comment (rtretyak) [inline src-java/testing/performance-tests/src/test/groovy/org/openkilda/performancetests/spec/benchmark/FlowDumpSpec.groovy:45]: doesn't really matter, but not a real check here. Needs an explicit `assert` keyword when inside 'each'

**Review by rtretyak — APPROVED** [review]

File: src-java/testing/performance-tests/src/test/groovy/org/openkilda/performancetests/spec/benchmark/FlowDumpSpec.groovy (hunk 1 above)
Comment (sergii-iakovenko) [reply src-java/testing/performance-tests/src/test/groovy/org/openkilda/performancetests/spec/benchmark/FlowDumpSpec.groovy]: Nice catch. Fixed.

File: src-java/testing/performance-tests/src/test/groovy/org/openkilda/performancetests/spec/benchmark/ConcurrentFlowCreateSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Unroll
+    def "Flow creation (concurrent) on mesh topology"() {
+        given: "A mesh topology"
+        def topo = new TopologyBuilder(flHelper.fls,
+                preset.islandCount, preset.regionsPerIsland, preset.switchesPerRegion).buildMeshes()
+        topoHelper.createTopology(topo)
+        flowHelperV2.setTopology(topo)
+
+        when: "A source switch"
```
Comment (andriidovhan) [inline src-java/testing/performance-tests/src/test/groovy/org/openkilda/performancetests/spec/benchmark/ConcurrentFlowCreateSpec.groovy:28]: as for me would be better to move it to the `and` block
for example:
```
given: "A mesh topology"
and: "A source switch for the flows"
when: "Create flows"
```

**Review by andriidovhan — APPROVED** [review]

File: src-java/testing/performance-tests/src/test/groovy/org/openkilda/performancetests/spec/benchmark/ConcurrentFlowCreateSpec.groovy (hunk 2 above)
Comment (sergii-iakovenko) [reply src-java/testing/performance-tests/src/test/groovy/org/openkilda/performancetests/spec/benchmark/ConcurrentFlowCreateSpec.groovy:28]: Make sense. Moved.

File: src-java/testing/performance-tests/src/main/groovy/org/openkilda/performancetests/helpers/TopologyHelper.groovy [hunk 3]
```diff
@@ -98,31 +101,36 @@ class TopologyHelper extends org.openkilda.functionaltests.helpers.TopologyHelpe
      * Delete the lab. Wait until topology is failed and delete all ISLs and all switches related to given
      * topology definition.
      */
-    def purgeTopology(TopologyDefinition topo, LabInstance lab = null) {
+    def purgeTopology(TopologyDefinition topo = null, LabInstance lab = null) {
```
[the remark this answers is not in the document]
Comment (sergii-iakovenko) [reply src-java/testing/performance-tests/src/main/groovy/org/openkilda/performancetests/helpers/TopologyHelper.groovy:103]: No, I'd like to revert this and move the flow cleanup code back to BaseSpecification.

**Comment by sergii-iakovenko** [comment]
Tested on the staging env. - no noticeable changes in opentsdb topics (rates, lags, etc.)

**Pull request merged**